aws/asg.go

```diff
@@ -260,6 +260,25 @@ func describeTargetGroups(elbv2svc elbv2iface.ELBV2API) (map[string]struct{}, er
 	return targetGroups, err
 }
 
+// map the target group slice into specific types such as instance, ip, etc
+func categorizeTargetTypeInstance(elbv2svc elbv2iface.ELBV2API, allTGARNs []string) (map[string][]string, error) {
+	targetTypes := make(map[string][]string)
+	err := elbv2svc.DescribeTargetGroupsPagesWithContext(context.TODO(), &elbv2.DescribeTargetGroupsInput{},
+		func(resp *elbv2.DescribeTargetGroupsOutput, lastPage bool) bool {
+			for _, tg := range resp.TargetGroups {
+				for _, v := range allTGARNs {
+					if v != aws.StringValue(tg.TargetGroupArn) {
+						continue
+					}
+					targetTypes[aws.StringValue(tg.TargetType)] = append(targetTypes[aws.StringValue(tg.TargetType)], aws.StringValue(tg.TargetGroupArn))
+				}
+			}
+			return true
```

Comment:
Not sure but I think you want to return `lastPage` here, because of the pagination possibility. 
What if the listing is more than a single page, then it will stop here if you return true IIRC.

Reply:
As far as I see we also do it wrong in `describeTargetGroups` @AlexanderYastrebov .

Reply:
>  To stop iterating, return false from the fn function.

I assume simply returning true iterates through all pages.

Reply:
not documented but from the sources its clearer: https://github.com/aws/aws-sdk-go/blob/main/service/elbv2/api.go#L2183